Improve update translations stats

SimonHeimberg opened the pull request:
**Type of change**
Enhancement

**Issue(s) closed**
Fixes -

**New behavior**
- data/i18n/translation_stats.conf has a comment header: 6469c426c05dd0392066e850f489619b7dd246f6
- on error, no other error is raises: 14c9c3abc235d4141448e25eb875798b3cc51edb

and code cleanup
- remove useless code: cf1936f6d4a2ea03feda2409c63d18d2fdf66aed
- line breaks in nicer places, : 6469c426c05dd0392066e850f489619b7dd246f6
- ...

**Additional context**
There will be a pr speeding up the script by calling pocount once with all directories: you can preview it at [branch update_translations_stats_speedup](https://github.com/SimonHeimberg/widelands/commits/update_translations_stats_speedup)
My intention of doing two steps is easier review, but maybe this makes it more complicated. Then I will adapt.

SimonHeimberg committed:
update_translation_stats.py only imports subprocess once

SimonHeimberg committed:
update_translation_stats.py writes comment to generated file

I checked the cc code (src/io/profile.cc), it skipps the comment.

SimonHeimberg committed:
remove useless if in update_translation_stats.py

subprocess.run(..., check=True) never returns a returncode != 0,
it throws subprocess.CalledProcessError then.

SimonHeimberg committed:
cleanup update_translation_stats.py

- line breaks in nicer place (not inside array index)
- process args as array instead of shell=True
- sorted() takes an iterable as input
- mode for open() without + because there is no reading
- locale_stats is a defaultdict, use its automatic generated entry

SimonHeimberg committed:
fix print() call in update_translation_stats.py

Fix it to what it was maybe ment.

- all the string is inside print(...), failed before: print(...) + str2
- .split() really splits, before it was limited to 0 times

SimonHeimberg committed:
update_translation_stats.py writes file without calling strip before

Add the extra newline with the section header,
there is none to remove at the end.

This saves a comment and maybe some storage space.

klaus-halfmann reviewed (commented):
python utils/update_translation_stats.py works fine after installing 
`brew install translate-toolkit`

SimonHeimberg committed:
Revert "remove useless if in update_translation_stats.py"

This reverts commit cf1936f6d4a2ea03feda2409c63d18d2fdf66aed.
It is unrelated to this branch.

SimonHeimberg committed:
update_translation_stats.py writes all output in case of error

as requested by tothxa

SimonHeimberg committed:
cleanup, update_translation_stats.py with shorter lines and proper indentation

tothxa reviewed (commented):
In addition to the inline comments:

 - `while` loop in line 89 should be replaced by `for column_counter in range(len(header_entries)):`
 - can't we drop the python version check from lines 148-150 now that we specify python3 in the `#!` line?

SimonHeimberg committed:
address review in update_translation_stats.py

use += to append to a string
for instead of while with fixed size

as suggested by tothxa

tothxa reviewed (commented):
Thank you.

SimonHeimberg committed:
use for i, v in enumerate() in update_translation_stats.py

as pylint recommends

SimonHeimberg committed:
update_translation_stats.py reports when column not found

SimonHeimberg committed:
update_translation_stats.py lets python print a traceback

SimonHeimberg committed:
update_translation_stats.py simply reads header line

I have not found any occurence where pocount writes the header on multiple lines.

In case it really happens there will be a clear error message since a few commits.

SimonHeimberg committed:
'utils/update_translation_stats.py' was automatically formatted.

SimonHeimberg committed:
put quotes round column name in update_translation_stats.py

Co-authored-by: Tóth András <80712255+tothxa@users.noreply.github.com>

SimonHeimberg committed:
'utils/update_translation_stats.py' was automatically formatted.

SimonHeimberg committed:
update_translation_stats.py reads with csv reader

instead of using own reader code. This simplifies error reporting as well.

tothxa reviewed (approved):
Thank you. Using the csv module was clever. :)

I wonder whether we could drop some of the paranoid checks in the follow-up, but for now we can keep to the original functionality.

SimonHeimberg committed:
rename variable in update_translation_stats.py

as suggested in review

SimonHeimberg committed:
use string literal concatenation instead of multiline string in update_translation_stats.py

for nicer indentation
while still being efficient

SimonHeimberg committed:
improve text in update_translation_stats.py

Co-authored-by: Tóth András <80712255+tothxa@users.noreply.github.com>

SimonHeimberg committed:
write forgoten newline in update_translation_stats.py

Co-authored-by: Tóth András <80712255+tothxa@users.noreply.github.com>

tothxa reviewed (approved):
Thank you!

Commit:
Merge branch 'master' of github.com:widelands/widelands into update_translations_stats
